pkgdb/tests/scrape.bats

```diff
@@ -162,6 +162,31 @@ setup_file() {
   assert_output '0'
 }
 
+# ---------------------------------------------------------------------------- #
+
+# Parallel scrapes are likely to run into concurrency issues when
+# the required database does not yet exist.
+# Multiple processes may try to create the database "at the same time",
+# in the past leading to failures.
+# bats test_tags=scrape:parallel
+@test "'pkgdb scrape' doesn't crash when run in parallel" {
+  # We attempt a parallel scrape of 8 parallel scrapes of the same package set
+  # for a total of 5 times.
+  # <ultiple runs increase the chance of concurrent events,
+  # the whole test is however a probabilistic measure.
+  for i in $(seq 5); do
+    # We don't want other tests polluting parallel test runs so we do this test
+    # with a unique cache directory.
+    run --separate-stderr sh -c '
+    PKGDB_CACHEDIR="$(mktemp -d)" parallel --halt-on-error 2 \
+      "echo {};
+       pkgdb scrape github:nixos/nixpkgs/ab5fd150146dcfe41fda501134e6503932cc8dfd \
+          legacyPackages '$NIX_SYSTEM' 'akkoma-emoji'" \
+      ::: $(seq 8)'
+    assert_success
+  done
+}
+
```

Comment:
question nonblocking: does the test in search.bats cover anything this doesn't?

Reply:
This was originally derived from the search test.
I don't know enough about the search code paths to say whether they might test something else in addition.

Reply:
the one difference here is that there is no `sleep` for any of the parallel runs.

Reply:
@zmitchell can correct me if I'm wrong, but I think we probably only need one of them

Reply:
The `sleep` is there to make sure that the search processes weren't all created at _exactly_ the same time, because that's synchronization hard mode. Doing this without sleeps could be flakey.

Reply:
I think we only need one of these tests, so I agree with @mkenigs

Reply:
alright, for the sake of this piece of code being old news anyway in medium timeframe, I'll be backing this out
I'd say a test that tests concurrent initialization of databases should concurrently initialize databases but alas.